Frameworks/UIKit/UIScrollView.mm

```diff
@@ -894,7 +894,11 @@ - (void)_refreshOrigin {
     [self setNeedsLayout];
 
     if ([self.delegate respondsToSelector:@selector(scrollViewDidScroll:)]) {
-        [self.delegate scrollViewDidScroll:self];
+        // NOTE: We have to push this off of the ScrollViewer's callback thread to work around
+        // issue #1865.
+        dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I'm not positive this is the best solution, but I wanted to get it out for review.  I think spot-fixing apps should be the last resort solution to any problem.  A proper fix from the Xaml team would be ideal, but that would require everyone to be running/testing on latest RS2 builds.  Also, if we do move back to on-thread scrolling, this issue would go away.  Let's synch up in person with Terry after we hear back from the Xaml team.

---
In reply to: [98736679](https://github.com/Microsoft/WinObjC/pull/1877#discussion_r98736679) [](ancestors = 98736679)